### PR TITLE
Bug 1191731 - Update to use new poster API to ensure created after vi…

### PR DESCRIPTION
**Pull request by aosmond**
…deo recorded.

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4e5d5e1ca0e7dbe3a9a846d1b5780b99c4ef7e31

File: apps/camera/js/controllers/storage.js [hunk 1]
```diff
@@ -143,8 +143,24 @@ StorageController.prototype.storePicture = function(picture) {
...
+  poster.filepath = video.filepath.replace('.3gp', '.jpg');
   video.isVideo = true;
-  this.app.emit('newmedia', video);
+
+  this.storage.addPicture(
+    poster.blob,
+    { filepath: poster.filepath },
+    function(error, path, absolutePath, fileBlob) {
+      // Replace the memory-backed Blob with the DeviceStorage file-backed File.
+      // Note that "video" references "poster", so video previews will use this
```
Comment (justindarc) [inline apps/camera/js/controllers/storage.js]: Do you mean that `video` *has a reference to* `poster`?

File: apps/camera/js/lib/camera/camera.js [hunk 2]
```diff
@@ -904,9 +904,7 @@ Camera.prototype.startRecording = function(options) {
       }
 
       video.filepath = filepath;
-      video.poster.filepath = filepath.replace('.3gp', '.jpg');
-      config.posterFilepath = video.poster.filepath;
-      config.posterStorageArea = self.storage.picture;
+      config.createPoster = true;
```
Comment (justindarc) [inline apps/camera/js/lib/camera/camera.js]: Why don't you just put this in `config` when you define it above?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=93c5f0e2b0fddb494cd852df4fb69956cded878c

**Pull request merged**